src/unk_02054884.c

```diff
@@ -175,41 +175,41 @@ void Party_GiveChampionRibbons(Party *party)
     }
 }
 
-int sub_02054B04(Party *param0, u16 param1)
+int Pokemon_DoPoisonDamage(Party *party, u16 param1)
 {
-    int v0 = 0;
-    int v1 = 0;
-    int v2, v3;
-    Pokemon *v4;
+    int numPoisoned = 0;
+    int numFainted = 0;
+    int i, partyCount;
+    Pokemon *mon;
 
-    v3 = Party_GetCurrentCount(param0);
+    partyCount = Party_GetCurrentCount(party);
 
-    for (v2 = 0; v2 < v3; v2++) {
-        v4 = Party_GetPokemonBySlotIndex(param0, v2);
+    for (i = 0; i < partyCount; i++) {
+        mon = Party_GetPokemonBySlotIndex(party, i);
 
-        if (Pokemon_CanBattle(v4)) {
-            if (Pokemon_GetValue(v4, MON_DATA_STATUS_CONDITION, NULL) & (MON_CONDITION_TOXIC | MON_CONDITION_POISON)) {
-                u32 v5 = Pokemon_GetValue(v4, MON_DATA_CURRENT_HP, NULL);
+        if (Pokemon_CanBattle(mon)) {
```

Comment:
**suggestion:** I think you can collapse these two `if` statements into one with a conjunction and it'll still match.